src/vorta/views/extract_dialog.py

```diff
@@ -616,7 +616,7 @@ def set_checkstate_recursively(self, index: QModelIndex, value: Qt.CheckState):
 
         item = index.internalPointer()
         for i in range(number_children):
-            child = index.child(i, 0)
+            child = index.sibling(i, 0)
```

Comment:
`sibling` is not the same as `child`. This is the reason for the one of the test failures.

Reply:
`child` is deprecated now. The docs recommend to use `QAbstractItemModel.index(row, column[, parent=QModelIndex()])` but when I use it, it gives
```
TypeError: index(self, row: int, column: int, parent: QModelIndex = QModelIndex()): first argument of unbound method must have type 'QAbstractItemModel'
```

Reply:
Use
```suggestion
            child = self.index(i, 0, index)
```

Reply:
Yeah it fixed it. Thanks!

Reply:
Should we add the color changing icons after PyQt6 upgrade? So that we won't need to update it to fix for PyQt6 afterwards.

Reply:
Some testing might be required for the PyQt6 upgrades, so maybe merging that and updating this branch afterwards would be better.

Reply:
It won't be that long before we could push this update I think. But either way I've a PR opened for the current fix and I might also change it to make it compatible with the PyQt6 as it is without requiring much change in future.

Reply:
There are many other PRs open, too. Updating this PR would be better than updating all of them to be compatible with pyqt6. Anyway, this is a bigger fix than that one and can wait on another branch as Manu said.

Reply:
Yeah, I made the required chances to the Icons PR. That can now be added anytime before or after the PyQt6 upgrade. It works on both, just requires to change the imports which this PR would be doing to all the imports anyway.

Reply:
In most cases, there shouldn't be many conflicts with existing PRs after merging the qt6 update. Since most changes are just to imports. And I prefer to be done with this update and have future work build on top of it instead of maintaining 2 versions for too long. And we merged many finished PRs for the next release.

So I'll do the 0.8.11 release and add a new 0.8 branch that will only get bugfixes.

The master branch will get this qt6 PR and the version 0.9.x in the future.

Will look into the failing tests now.